hilti/toolchain/src/compiler/codegen/types.cc

```diff
@@ -845,9 +845,9 @@ struct VisitorTypeInfoDynamic : hilti::visitor::PreOrder<cxx::Expression, Visito
                 accessor = fmt(", hilti::rt::type_info::struct_::Field::accessor_optional<%s>()",
                                cg->compile(f.type(), codegen::TypeUsage::Storage));
 
-            fields.push_back(fmt("hilti::rt::type_info::struct_::Field{ \"%s\", %s, offsetof(%s, %s)%s }",
+            fields.push_back(fmt("hilti::rt::type_info::struct_::Field{ \"%s\", %s, offsetof(%s, %s), %s%s }",
                                  cxx::ID(f.id()), cg->typeInfo(f.type()), cxx::ID(*n.typeID()), cxx::ID(f.id()),
-                                 accessor));
+                                 f.isInternal().has_value(), accessor));
```

Comment:
That looks like a problem with the `isInternal()` API, it should really just return a boolean (instead of `auto`ing the AttributeSet::find()` result`. You could just make that change as part of this commit if you want; also for the other `is*` methods in `Field`. Or we can do some other time.